text/0062-recoverable-scis.md

```diff
@@ -0,0 +1,112 @@
+- Feature Name: recoverable_scis
+- Start Date: 2023-03-02
+- MCIP PR: [mobilecoinfoundation/mcips#0062](https://github.com/mobilecoinfoundation/mcips/pull/0062)
+- Tracking Issue: (link to tracking issue)
+
+# Summary
+
+[summary]: #summary
+
+This MCIP proposes that inputs to be used in an SCI first be sent to a reserved subaddress `SCI_SUBADDRESS_INDEX`, extending the scheme from [MCIP #36](0036-reserved-subaddresses.md)
+
+This MCIP also proposes that new RTH memos be created to further support inputs used in SCIs, extending the scheme from [MCIP #4](0004-recoverable-transaction-history.md)
+
+# Motivation
+
+[motivation]: #motivation
+
+Similar to the motivations in [MCIP #32](0032-fog-compatible-gift-codes.md), a concern for users is the ability to track things across linked devices using the blockchain (and not separate services). It is a concern wrt txos used as an SCI in that one instance of an account could use a txo as an SCI and before the recipient of said SCI completes a transaction with it (which could take some arbitrary yet non trivial amount of time), another (or even the same, if data is not properly tracked) instance of the account could use that txo as an input in another transaction, therefore unintentionally rendering the SCI cancelled.
+
+Another concern is that because there are multiple types of ways to specify the rules of an SCI (one example being [MCIP #42](0042-partial-fill-rules.md) which extended the SCI rules to include partial fills) one might want to have some record of that information on the blockchain (instead of relying on a secondary service)
+
+# Guide-level explanation - WIP (awaiting feedback and discussion)
+
+[guide-level-explanation]: #guide-level-explanation
+
+When one wishes to generate an SCI, the app must first make a self payment to the `SCI_SUBADDRESS_INDEX = u64::MAX - 3`, which is now reserved for `TxOut`s used as `SCI`s.
+Apps should not select `TxOut`s at this subaddress for inputs in other transactions, and they may choose not to include these in the account balances as they represent the amount of active `SCI`s.
+
+The `TxOut` generated for use as the SCI must contain an `SCI Generation Memo` to identify the details of the SCI it was included in (specifics TBD as feature evolves from feedback and discussion)
+
+After an SCI is sent, an app can check if it has been claimed based on if this `TxOut` has been spent. Until it has been spent, the app can potentially cancel the `SCI` by sending a self payment with the `TxOut`.
+
+We additionally extend Recoverable Transaction History memos from [MCIP #4](0004-recoverable-transaction-history.md) to handle `SCI`s as well.
+
+| Memo type bytes | Name                  |
+| --------------- | --------------------- |
+| 0x0301          | SCI generation memo   |
+| 0x0302          | SCI cancellation memo |
+| 0x0003          | SCI sender memo       |
```

Comment:
Do we need a sender memo? We're imagining these scis are mostly traded around via the deqs, so it is unlikely that the identify of the sci creator is known to the sci executor.

Reply:
It's worth mentioning that the consumer of the SCI, who includes a TxOut to pay back the originator change as part of a partial fill, cannot alter the memo, so it is up to the originator to put something there.
There should probably be a memo type that indicates this is a change or partial change - there could be required change input rule (represented as a required output) and/or a partial change input rule.
I'm not sure what information we'd want to include in that memo. Some things we might consider putting there are the public key of the original TxOut, and maybe for partial change the max amount that was available?